lib/buildTree.nix

```diff
@@ -59,7 +69,15 @@ let
       then pkg.bin or pkg.out or pkg
       else pkg;
 
-    getExe = x: "${lib.getBin x}/bin/${x.meta.mainProgram or (lib.getName x)}";
+    getExe =
+      let
+        isDir = p: builtins.pathExists (toString p + "/.");
+      in
+      x:
+      lib.findFirst (p: builtins.pathExists p && !isDir p) "" [
+        "${lib.getBin x}/bin/${x.meta.mainProgram or (lib.getName x)}"
+        (lib.getBin x)
+      ];
```

Comment:
This is departing from the `nix run` logic. Is it not enough to fix the mainProgram?

Reply:
Does this not do import from derivation? This might not evaluate in some CIs or in flakes (without flag).